moda-contracts/src/interfaces/ICatalog.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+/**
+ * @title IRegistry
+ * @dev Interface for Registry
+ */
+interface ICatalog {
+    /**
+     * @dev Status of a registered track
+     */
+    enum TrackStatus {
+        PENDING,
+        VALIDATED,
+        INVALIDATED
+    }
+
+    /**
+     * @dev Represents a registered track
+     * TrackStatus - The status of the track (pending, validated, invalidated)
+     * TrackOwner - The address that registered the track
+     * TrackRegistrationHash - This is a IPFS hash of the track metadata
+     * FingerprintHash - This is a IPFS hash of the track fingerprint
+     * ValidationHash - This is a hash of proof of validation data
```

Comment:
I wonder if we should provide details of who the verifier was? AudioMagic could be one of many. Or maybe it is an Organization? 

Should we add `address trackValidator`? Maybe trusted validators are listed in the ModaRegistry?

Reply:
So I have the verifier role, which are addresses that can update a tracks status, these are currently in the ModaRegistry as an enumerable set using AccessControlEnumerable, are those the addresses that you are referring to when you say `address trackValidator` ?

I have added trackVerifier as a field in the struct so we can easily pull up who verified a particular track.

I think i need to talk to you more about the verifier role, Track Verifier contract etc to get a clear understanding of how that will work.